modules/apps/image/image-api/src/main/java/com/liferay/image/company/provider/ImageCompanyIdUpgradeProvider.java

```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.image.company.provider;
+
+import com.liferay.portal.kernel.dao.orm.ActionableDynamicQuery;
+
+import java.sql.Connection;
+
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import org.osgi.annotation.versioning.ProviderType;
+
+/**
+ * @author Alicia Garcia
+ */
+@ProviderType
+public interface ImageCompanyIdUpgradeProvider<T> {
```

Comment:
Hi @AliciaGarciaGarcia,

If you expose the whole upgrade class, you can directly reuse it without wrapping it. That's how [UpgradeDiscussionSubscriptionClassName](https://github.com/liferay/liferay-portal/blob/8072955ec68042ea0ae85320e6c3cbab52cb57fc/modules/apps/comment/comment-page-comments-web/src/main/java/com/liferay/comment/page/comments/web/internal/upgrade/PageCommentsWebUpgrade.java#L60-L66) works. 

What do you think?

Reply:
https://github.com/liferay-lima/liferay-portal/pull/2135